src/main/java/io/github/reconsolidated/zpibackend/domain/item/Item.java

```diff
@@ -69,22 +73,41 @@ public Item(Store store, ItemDto itemDto) {
                                     itemDto.getSchedule().getStartDateTime() :
                                     itemDto.getSchedule().getEndDateTime(),
                             ReservationType.NONE)));
-            this.initialSchedule = this.schedule;
+            this.initialSchedule = new Schedule(this,
+                    List.of(new Availability(
+                            itemDto.getSchedule().getStartDateTime(),
+                            itemDto.getSchedule().getEndDateTime() == null ?
+                                    itemDto.getSchedule().getStartDateTime() :
+                                    itemDto.getSchedule().getEndDateTime(),
+                            ReservationType.NONE)));
         }
-        this.subItems = itemDto.getSubItems();
+        this.subItems = itemDto.getSubItems().stream().map(SubItem::new).toList();
         this.reservations = new ArrayList<>();
     }
 
     public SubItemListDto getSubItemsListDto() {
-        ArrayList<SubItemDto> subItemsDto = new ArrayList<>();
+        ArrayList<SubItemInfoDto> subItemsDto = new ArrayList<>();
         for (SubItem subItem : subItems) {
-            subItemsDto.add(subItem.toSubItemDto());
+            subItemsDto.add(subItem.toSubItemInfoDto());
         }
         return new SubItemListDto(subItemsDto);
     }
 
-    public SubItemDto toSubItemDto() {
-        return new SubItemDto(itemId, title, subtitle);
+    public SubItemInfoDto toSubItemDto() {
+        return new SubItemInfoDto(itemId, title, subtitle);
     }
 
+    public boolean isFixedPast() {
+        if (store.getStoreConfig().getCore().getFlexibility()) {
+            return false;
+        }
+        if (store.getStoreConfig().getCore().getPeriodicity()) {
+            return !subItems.stream().anyMatch(subItem -> subItem.getStartDateTime().isAfter(LocalDateTime.now()));
+        } else {
+            return schedule != null &&
+                    !schedule.getAvailableScheduleSlots().isEmpty() &&
+                    schedule.getAvailableScheduleSlots().get(0).getStartDateTime().isBefore(LocalDateTime.now());
```

Comment:
are you sure that 0th element is earliest too? shouldn't we iterate and find earliest one?

Reply:
in case of fixed items it is the only slot in schedule